Tests/UnitTests/FileShareClientTests/BatchAttributeSearchTests.cs

```diff
@@ -152,13 +152,43 @@ public void TestNameOfAndAttributeValuesinToStringMethod()
             var colourList = new List<string> { "red", "blue" };
 
             var searchBatchAttributes = new BatchAttributesSearchAttribute
-                    {
-                        Key = "Colour",Values = colourList
+            {
+                Key = "Colour",
+                Values = colourList
             };
             var attributeValues = searchBatchAttributes.ToString();
             Assert.AreEqual("class BatchAttributesSearchAttribute {\n Key: Colour\n Values: red, blue\n}\n", attributeValues);
         }
 
+        #region BatchSearch with MaxAttributeValueCount
+
+        [Test]
+        public async Task DoesBatchAttributeSearchReturnsSucessWithMaxAttributeValueCountandFilter()
```

Comment:
Please add [TestCase]s to show maxAttributeValueCount is set correctly for various values, not just 2

````
        [TestCase(-1)]
        [TestCase(0)]
        [TestCase(2)]
        [TestCase(10)]
        [TestCase(1000)]
        public async Task DoesBatchAttributeSearchReturnsSucessWithMaxAttributeValueCountandFilter(int maxAttributeValueCount)
        {
            var response = await fileShareApiClient.BatchAttributeSearch("$batch(key) eq 'value'", maxAttributeValueCount, cancellationToken: CancellationToken.None);
            Assert.AreEqual("/basePath/attributes/search", lastRequestUri.AbsolutePath);
            Assert.AreEqual($"?$filter=$batch(key)%20eq%20%27value%27&maxAttributeValueCount={maxAttributeValueCount}", lastRequestUri.Query);
            Assert.IsTrue(response.IsSuccess);
        }

````